### DISCUSSION TITLE
[Feature request] Add "os/arch" and "sub-container id" to response of "/users/{username}/packages/{package_type}/{package_name}"

### DISCUSSION THREAD

**Discussion by ManiMatter** [community/community#120087] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

As outlined [here](https://github.com/orgs/community/discussions/119987), there seems to be no way to establish the link between a multi-architecture container and it's respective sub-containers via github API.

The most logical way, in my opinion, would be to add the information to the response of ["/users/{username}/packages/{package_type}/{package_name}"](https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#get-a-package-for-a-user).

Example:
Currently, querying https://api.github.com/users/ManiMatter/packages/container/decluttarr/versions/204488276 returns:
```
{
    id: 204488276,
    name: 'sha256:44a623848836eb4629e22849eab526fffe4eb823d43a94aa1aef70d3dbed3c2a',
    url: 'https://api.github.com/users/ManiMatter/packages/container/decluttarr/versions/204488276',
    package_html_url: 'https://github.com/users/ManiMatter/packages/container/package/decluttarr',
    created_at: '2024-04-16T16:32:06Z',
    updated_at: '2024-04-16T16:32:06Z',
    html_url: 'https://github.com/users/ManiMatter/packages/container/decluttarr/204488276',
    metadata: { package_type: 'container', container: { tags: ['latest', 'dev'] } }
} 
```

What I propose is that the metadata (last row) gets enhanced to this:

```
    metadata: { package_type: 'container', container: { tags: ['latest', 'dev'], sub_containers: [{id: 204488239, os_arch: 'linux/amd64'}, {id: 204488246, os_arch: 'linux/arm64'}, {id: 204488256, os_arch: 'unknown/unknown'}] } }
```

In other words, the proposal is that the package id of the sub containers and their respective architecture are added to the metadata section.

This way, via API one can arrive at the same information as via the website under this [URL (tab "OS/Arch") ](https://github.com/ManiMatter/decluttarr/pkgs/container/decluttarr/204488276)

**Comment by ManiMatter** [comment]
Anyone?

**Comment by ManiMatter** [comment]
Closing this since no response.
